hazelcast/src/main/java/com/hazelcast/nio/tcp/nonblocking/NonBlockingReadHandler.java

```diff
@@ -145,14 +145,17 @@ public void requestMigration(NonBlockingIOThread newOwner) {
 
     @Override
     public void handle() {
-        eventCount.inc();
-        lastReadTime = Clock.currentTimeMillis();
-        if (!connection.isAlive()) {
-            String message = "We are being asked to read, but connection is not live so we won't";
-            logger.finest(message);
-            return;
-        }
         try {
+            eventCount.inc();
+            // we are going to set the timestamp even if the socketChannel is going to fail reading. In that case
+            // the connection is going to be closed anyway.
+            lastReadTime = Clock.currentTimeMillis();
```

Comment:
the tricky try/catch stuff is gone thanks to this PR. 
I'd still prefer the version things like checking/logging was placed outside the try/catch block... anyway, it's not terribly important.